apps/gitness/src/pages-v2/pull-request/__tests__/pull-request-utils.test.ts

```diff
@@ -0,0 +1,253 @@
+    const result = processReviewDecision(mockReviewDecision, mockReviewedSHA, mockSourceSHA)
+    expect(result).toBe(PullReqReviewDecision.outdated)
+  })
+})
+
+describe('generateStatusSummary', () => {
+  it('should generate a summary of status checks', () => {
+    const result = generateStatusSummary(mockChecks)
+    expect(result.summary.total).toBe(2)
+    expect(result.summary.successReq).toBe(1)
+    expect(result.summary.failed).toBe(1)
+  })
+})
+
+// Mock data for determineStatusMessage
+const mockChecksForStatusMessage = [
+  { check: { status: 'success' as EnumCheckStatus }, required: true },
+  { check: { status: 'failure' as EnumCheckStatus }, required: false }
+]
+
+describe('determineStatusMessage', () => {
+  it('should determine the status message based on the status summary', () => {
+    const result = determineStatusMessage(mockChecksForStatusMessage as TypeCheckData[])
+    expect(result).toEqual({
+      color: 'text-success',
+      content: '1 failed, 1 succeeded',
+      status: 'success',
+      title: 'All required checks passed'
+    })
+  })
+})
+
+describe('extractSpecificViolations', () => {
+  it('should extract specific violations based on rule', () => {
+    const result = extractSpecificViolations(mockViolationsData, 'rule1')
+    expect(result).toEqual([{ code: 'rule1', description: 'Violation 1' }])
+  })
+})
+
+describe('checkIfOutdatedSha', () => {
+  it('should return true if SHAs differ', () => {
+    const result = checkIfOutdatedSha(mockReviewedSHA, mockSourceSHA)
+    expect(result).toBe(true)
+  })
+})
+
+describe('capitalizeFirstLetter', () => {
+  it('should capitalize the first letter of a string', () => {
+    const result = capitalizeFirstLetter('hello')
+    expect(result).toBe('Hello')
+  })
+})
+
+describe('findChangeReqDecisions', () => {
+  it('should find change request decisions from entries', () => {
+    const result = findChangeReqDecisions(mockEntriesForChangeReq, 'approve')
+    expect(result).toEqual([
+      {
+        owner_evaluations: [{ review_decision: 'approve' }]
+      },
+      {
+        owner_evaluations: [{ review_decision: 'approve' }]
+      }
+    ])
+  })
+})
+
+describe('findWaitingDecisions', () => {
+  it('should find waiting decisions from entries', () => {
+    const result = findWaitingDecisions(mockEntriesForWaitingDecisions)
+    expect(result).toEqual([])
+  })
+})
+
+describe('normalizeGitFilePath', () => {
+  it('should normalize file paths with spaces', () => {
+    const result = normalizeGitFilePath('path/with space/file.txt')
+    expect(result).toBe('path/with space/file.txt')
+  })
+})
+
+describe('generateAlphaNumericHash', () => {
+  it('should generate an alphanumeric hash of specified length', () => {
+    const result = generateAlphaNumericHash(5)
+    expect(result).toHaveLength(5)
+  })
+})
+
+describe('getErrorMessage', () => {
+  it('should return an error message from an error object', () => {
```

Comment:
Looks like the function can extract from a few places... some tests to show that would be great

Reply:
/canary/apps/gitness/src/utils/__tests__/error-utils.test.ts
removed this test since it was duplicated